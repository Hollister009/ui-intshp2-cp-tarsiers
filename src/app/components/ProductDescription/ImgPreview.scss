@import '../../../styles/mixin';
@import '../../../styles/variables';

.imgpreview {
  border: 1px solid black;
}
