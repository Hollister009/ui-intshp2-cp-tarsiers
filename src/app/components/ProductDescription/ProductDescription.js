import React, { Component } from 'react';
import appConfig from '../../../config/appConfig';

import styles from './ProductDescription.module.scss';

class ProductDescription extends Component {
  constructor(props) {
    super(props);
    this.state = { quantity: 0 };
  }

  increment = () => {
    this.setState(prevState => ({ quantity: prevState.quantity + 1 }));
  };

  decrement = () => {
    const { quantity } = this.state;

    if (quantity > 0) {
      this.setState(prevState => ({ quantity: prevState.quantity - 1 }));
    }
  };

  render() {
    const { item } = this.props;
    const { quantity } = this.state;

    if (!item) {
      return null;
    }
    const price = quantity > 0 ? item.price * quantity : item.price;
    const sizes = item.sizes.map((element, index, array) => (
      <React.Fragment>
        <a href="/" className={styles.size} onClick={e => e.preventDefault()}>
          {element}
        </a>
        {index + 1 !== array.length ? <span>-</span> : null}
      </React.Fragment>
    ));

    return (
      <section className={styles.section}>
        <div className={styles.preview} />
        <div className={styles.details}>
          <div className={styles.info}>
            <h1 className={styles.title}>{item.title}</h1>
            <h3 className={styles.subtitle}>
              {appConfig.productDescription.subheader}
            </h3>
            <p className={styles.text}>
              {appConfig.productDescription.description}
            </p>
          </div>
          <div className={styles.select}>
            <div className={styles.flex_row}>
              <p className={styles.choose}>Choose Size</p>
              <div className={styles.sizes}>{sizes}</div>
            </div>
            <div className={styles.flex_row}>
              <p className={styles.choose}>Choose Quantity</p>
              <div className={styles.quantity}>
                <button type="button" onClick={this.increment}>
                  &#43;
                </button>
                <span>{quantity}</span>
                <button type="button" onClick={this.decrement}>
                  &#45;
                </button>
              </div>
            </div>
          </div>
          <div className={styles.order}>
            <p className={styles.price}>{`Price: ${price}$`}</p>
            <div className={styles.buttons_row}>
              <button type="button">
                <i className="fas fa-globe" />
              </button>
              <button type="button">
                <i className="fas fa-cart-plus" />
              </button>
              <button type="button">
                <i className="far fa-heart" />
              </button>
              <button type="button" className={styles.btn_order}>
                Order Now
              </button>
            </div>
          </div>
        </div>
      </section>
    );
  }
}

export default ProductDescription;
