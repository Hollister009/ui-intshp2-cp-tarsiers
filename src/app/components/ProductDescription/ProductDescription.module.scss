@import '../../../styles/mixin';
@import '../../../styles/variables';

.section {
  @include d-flex(center);

  p {
    font-size: 18px;
    font-family: $main-font-regular;
    font-weight: 400;
  }

  .flex_row {
    @include d-flex();
  }
  .choose {
    margin-right: 20px;
  }
  .price {
    font-weight: bold;
  }
}

.preview {
  width: 70%;
}

.details {
  margin: 0 40px;
  margin-top: 100px;

  .info {
    @include d-flex(flex-start, column);

    & > .title {
      color: $color-main;
      font-family: $main-font-bold;
      font-size: 52px;
      text-transform: uppercase;
      line-height: 45px;
    }
    & > .subtitle {
      color: $color-dark;
      font-family: $main-font-regular;
      font-weight: 400;
      font-size: 18px;
      margin: 30px 0 15px 0;
    }
    & > .text {
      color: $color-dark;
      font-family: $main-font-regular;
      font-size: 16px;
      line-height: 30px;
      margin: 15px 0 30px 0;
    }
  }

  .select {
    padding: 30px 0;
    border-top: 2px solid $color-light-2;
    border-bottom: 2px solid $color-light-2;
    @include d-flex();

    .sizes {
      @include d-flex() {
        align-items: center;
      }
    }

    .size {
      color: $color-dark;
      font-size: 18px;
      padding: 0 5px;
      text-decoration: none;
      text-transform: uppercase;
      // transition: 0.3s ease;
      outline: none;

      &:hover,
      &:focus-within {
        animation: scale 0.5s forwards ease;
        color: $color-main;
      }
    }
    .quantity {
      @include d-flex() {
        align-items: baseline;
      }

      button {
        background: none;
        font-size: 18px;
        padding: 0;

        &:first-of-type {
          padding-left: 10px;
        }

        &:last-of-type {
          padding-right: 10px;
        }
      }

      span {
        color: $color-main;
        font-size: 18px;
        padding: 0 10px;
      }
    }
  }

  .order {
    padding: 30px 0;
    @include d-flex();

    & > .buttons_row {
      @include d-flex();

      button {
        background: none;
        padding: 0 10px;
        color: $color-dark;
        font-size: 18px;

        &:hover {
          color: $color-main;
        }
      }

      .btn_order {
        color: $color-white;
        background-color: $color-main;
        border-radius: 30px;
        padding: 20px 30px;
        font-size: 16px;

        &:hover {
          background-color: $color-dark;
        }
      }
    }
  }
}

@keyframes scale {
  from {
    transform: scale(1);
  }
  to {
    transform: scale(1.1);
  }
}
