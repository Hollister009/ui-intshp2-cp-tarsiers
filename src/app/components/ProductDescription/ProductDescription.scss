@import '../../../styles/mixin';
@import '../../../styles/variables';

p {
  font-size: 18px;
  font-family: $main-font-regular;
  font-weight: 400;
}

.product-details {
  @include d-flex(center);
}

.images {
  width: 70%;
}

.product-description {
  @include d-flex(inherit, column);

  margin: 100px 40px 0 40px;
}

.product-text {
  @include d-flex(inherit, column);

  .product-header {
    color: $color-main;
    font-family: $main-font-bold;
    font-size: 52px;
    text-transform: uppercase;
    line-height: 45px;
  }

  .product-subheader {
    color: $color-dark;
    font-family: $main-font-regular;
    font-weight: 400;
    font-size: 18px;
    margin: 30px 0 15px 0;
  }

  .product-info {
    color: $color-dark;
    font-family: $main-font-regular;
    font-size: 16px;
    line-height: 30px;
    margin: 15px 0 30px 0;
  }
}

.size-quantity {
  @include d-flex();

  border-top: 2px solid $color-light-2;
  border-bottom: 2px solid $color-light-2;
  padding: 30px 0;

  .choose {
    margin-right: 20px;
  }

  .p-size {
    @include d-flex();

    .sizes {
      @include d-flex();

      .size {
        color: $color-dark;
        font-size: 18px;
        text-decoration: none;
        padding: 0 5px;
        text-transform: uppercase;

        &:hover {
          animation: 0.5s scale forwards;
          color: $color-main;

          @keyframes scale {
            from {
              transform: scale(1);
            }
            to {
              transform: scale(1.1);
            }
          }
        }

        &:focus-within {
          color: $color-main;
          font-size: 18px;
        }
      }
    }
  }

  .p-quantity {
    @include d-flex();

    .quantity {
      @include d-flex();

      align-items: flex-start;

      button {
        background: none;
        font-size: 18px;
        padding: 0 10px;
      }

      span {
        color: $color-main;
      }
    }
  }
}

.order-section {
  @include d-flex();

  padding: 30px 0;
  align-items: center;

  .price {
    font-weight: bold;
  }

  .order-buttons {
    @include d-flex();

    align-items: center;

    button {
      background: none;
      padding: 0 10px;
      color: $color-dark;
      font-size: 18px;

      &:hover {
        color: $color-main;
      }
    }

    .order-now {
      color: $color-white;
      background-color: $color-main;
      border-radius: 30px;
      padding: 20px 30px;
      font-size: 16px;

      &:hover {
        background-color: $color-dark;
      }
    }
  }
}
