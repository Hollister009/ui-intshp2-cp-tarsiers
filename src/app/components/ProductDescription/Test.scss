.img-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 103px;
  background-color: white;
  width: 40%;

  figure {
    width: 150px;
    position: relative;
    background-color: white;

    img {
      display: block;
      max-width: 180px;
      pointer-events: none;
    }

    div {
      width: 300px;
      height: 300px;
      background-repeat: no-repeat;
      background-size: 200% 200%;
      opacity: 0;
      position: absolute;
      top: -50px;
      right: -200px;
      border-radius: 50%;
      padding: 30px;
      border: 5px solid rgba(255, 255, 255, 0.5);
      &:hover {
        opacity: 0;
      }
    }

    &:hover div {
      opacity: 1;
    }
  }
}

.thumbnails-box {
  margin-top: 30px;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: 2;
  grid-column-gap: 10px;
  grid-row-gap: 15px;
  img {
    width: 70px;
    height: auto;
  }
}
