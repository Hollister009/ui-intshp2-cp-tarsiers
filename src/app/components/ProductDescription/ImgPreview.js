/* eslint-disable max-len */
/* eslint-disable react/prefer-stateless-function */
/* eslint-disable jsx-a11y/alt-text */
import React, { Component } from 'react';
// import ImgSwitch from './ImgSwitch'
// import ActiveThunb from './ActiveTumb';
import './Test.scss';

export default class ImgPreview extends Component {
  constructor(props) {
    super(props);
    const { item } = this.props;

    this.state = {
      figureStyle: {
        backgroundImage: `url(${item.src})`,
        backgroundPosition: '0% 0%'
      },
      mainImgURL: item.src
    };
  }

  handleMouseMove = e => {
    const { left, top, width, height } = e.target.getBoundingClientRect();
    const x = ((e.pageX - left) / width) * 100;
    const y = ((e.pageY - top) / height) * 100;

    this.setState(state => ({
      ...state,
      figureStyle: { ...state.figureStyle, backgroundPosition: `${x}% ${y}%` }
    }));
  };

  changeMainImage = e => {
    const newMainUrl = e.target.getAttribute('src');

    this.setState({
      mainImgURL: newMainUrl,
      figureStyle: { backgroundImage: `url(${newMainUrl})` }
    });
  };

  render() {
    const { item } = this.props;
    const { figureStyle } = this.state;
    const thumbnails = Object.values(item.colorUrls).map(el => (
      <img src={el} alt="" key={el} onClick={this.changeMainImage} />
    ));

    return (
      <div className="img-container">
        <figure onMouseMove={this.handleMouseMove}>
          <img src={this.state.mainImgURL} />
          <div style={figureStyle} />
        </figure>
        <div className="thumbnails-box">{thumbnails}</div>
      </div>
    );
  }
}
