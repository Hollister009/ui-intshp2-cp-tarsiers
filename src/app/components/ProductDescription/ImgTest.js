/* eslint-disable max-len */
/* eslint-disable react/prefer-stateless-function */
/* eslint-disable jsx-a11y/alt-text */
import React, { Component } from 'react';
import './Test.scss';

const src =
  'https://res.cloudinary.com/so/image/upload/v1548069002/products/reebok%20jacket/red.jpg';

export default class ImgTest extends Component {
  constructor() {
    super();
    this.state = {
      figureStyle: {
        backgroundImage: `url(${src})`,
        backgroundPosition: '0% 0%'
      }
    };
  }

  handleMouseMove = e => {
    const { left, top, width, height } = e.target.getBoundingClientRect();
    const x = ((e.pageX - left) / width) * 100;
    const y = ((e.pageY - top) / height) * 100;

    this.setState(state => ({
      ...state,
      figureStyle: {
        backgroundImage: `url(${src})`,
        backgroundPosition: `${x}% ${y}%`
      }
    }));
  };

  render() {
    const { figureStyle } = this.state;

    return (
      <div className="test-container">
        <figure onMouseMove={this.handleMouseMove} style={figureStyle}>
          <img src={src} />
        </figure>
      </div>
    );
  }
}
