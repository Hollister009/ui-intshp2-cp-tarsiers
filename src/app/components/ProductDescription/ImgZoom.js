/* eslint-disable max-len */
/* eslint-disable react/prefer-stateless-function */
/* eslint-disable jsx-a11y/alt-text */
import React, { Component } from 'react';
import './Test.scss';

// const src =
//   'https://res.cloudinary.com/so/image/upload/v1548069002/products/reebok%20jacket/red.jpg';

export default class ImgZoom extends Component {
  constructor(props) {
    super(props);

    const { item } = this.props;

    this.state = {
      figureStyle: {
        backgroundImage: `url(${item.src})`,
        backgroundPosition: '0% 0%'
      }
    };
  }

  handleMouseMove = e => {
    const { left, top, width, height } = e.target.getBoundingClientRect();
    const x = ((e.pageX - left) / width) * 100;
    const y = ((e.pageY - top) / height) * 100;

    this.setState(state => ({
      ...state,
      figureStyle: {
        backgroundImage: `url(${item.src})`,
        backgroundPosition: `${x}% ${y}%`
      }
    }));
  };

  render() {
    const { figureStyle } = this.state;

    return <div onMouseMove={this.handleMouseMove} style={figureStyle} />;
  }
}
