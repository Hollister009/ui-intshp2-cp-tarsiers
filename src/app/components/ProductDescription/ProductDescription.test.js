import React from 'react';
import { shallow } from 'enzyme';
import ProductDescription from './ProductDescription';

describe('<ProductDescription />', () => {
  it('should match its snapshot', () => {
    const wrapper = shallow(<ProductDescription />);

    expect(wrapper).toMatchSnapshot();
  });
});
