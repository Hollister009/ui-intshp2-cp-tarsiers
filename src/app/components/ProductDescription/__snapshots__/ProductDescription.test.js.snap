// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDescription /> should match its snapshot 1`] = `""`;
